flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicTableFactory.java

```diff
@@ -187,6 +188,9 @@ public DynamicTableSource createDynamicTableSource(Context context) {
 
     @Override
     public DynamicTableSink createDynamicTableSink(Context context) {
+        // validate the table option 'topic' and 'topic-pattern'
+        validateSinkTopic(Configuration.fromMap(context.getCatalogTable().getOptions()));
```

Comment:
Use `helper.getOptions()` (already stored in a variable `tableOptions` below) instead of building another `Configuration` here? Is there a reason to have this on the very top and not e.g. where `validateTableSinkOptions` is called?